metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -958,6 +962,10 @@ class MetalsLanguageServer(
         case Right(value) =>
           val old = userConfig
           userConfig = value
+          if (userConfig.excludedPackages != old.excludedPackages) {
+            userConfig.excludedPackages.foreach(excludedPackageHandler.update)
```

Comment:
so logic is inconsistent here now
if (new=None   !=   old=Some(a, b, c)) {
  don't update exclusions as we do that only for Some (foreach)
  but reindex classpath
}

Question what should happen on `None`? 
- treat it as 'reset' flag?
- treat it as 'keep previous' flag?

Reply:
Ah, yea good catch. My mistake. I changed this a bit to now no matter what call update when the value has changed. If there are new packages (`Some`), then we call `prepateCache` again with the new pkgs. If the value is now a `None`, we just assign the default exclusions to the cache.